contracts/staking/RewardDelegators.sol

```diff
@@ -411,4 +427,68 @@ contract RewardDelegators is
             }
         }
     }
+    
+    IEpochSelector public epochSelector;
+    event EpochSelectorUpdated(IEpochSelector newEpochSelector);
+    
+    function updateEpochSelector(IEpochSelector _epochSelector) onlyAdmin external{
+        _updateEpochSelector(_epochSelector);
+    }
+
+    function _updateEpochSelector(IEpochSelector _epochSelector) internal {
+        epochSelector = _epochSelector;
+        emit EpochSelectorUpdated(_epochSelector);
+    }
+
+    function sqrt(uint y) internal pure returns (uint z) {
+        if (y > 3) {
+            z = y;
+            uint x = y / 2 + 1;
+            while (x < z) {
+                z = x;
+                x = (y / x + x) / 2;
+            }
+        } else if (y != 0) {
+            z = 1;
+        }
+    }
+
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    bytes32 public immutable POND_TOKEN_ID;
+
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    bytes32 public immutable MPOND_TOKEN_ID;
+
+    uint256 private constant ONE_MILLION = 1_000_000;
+
+    uint256 public thresholdForSelection; // 0.5 million POND or equivalent (value is assigned in initializer)
+    event UpdateThresholdForSelection(uint256 newThreshold);
+
+    function updateThresholdForSelection(uint256 newThreshold) onlyAdmin external {
+        _updateThresholdForSelection(newThreshold);
+    }
+
+    function _updateThresholdForSelection(uint256 _newThreshold) internal {
+        thresholdForSelection = _newThreshold;
+        emit UpdateThresholdForSelection(_newThreshold);
+    }
+
+    event RefreshClusterDelegation(address cluster);
+    function refreshClusterDelegation(address[] calldata clusterList) onlyAdmin external {
+        for (uint256 index = 0; index < clusterList.length; index++) {
+            address cluster = clusterList[index];
+
+            uint256 totalDelegations = _getTotalDelegations(cluster);
+
+            if(totalDelegations >= thresholdForSelection){
+                epochSelector.insert(cluster, uint96(sqrt(totalDelegations)));
+            }
+
+            emit RefreshClusterDelegation(cluster);   
+        }
+    }
+
+    function _getTotalDelegations(address cluster) internal returns(uint256 totalDelegations){
+        totalDelegations = clusters[cluster].totalDelegations[POND_TOKEN_ID] + clusters[cluster].totalDelegations[MPOND_TOKEN_ID];
```

Comment:
Incorrect calculation, @akshay111meher recheck

Reply:
Also the threshold  is that cluster should have at least 0.5 MPond. POND balance doesn't matter